### PR TITLE
Add tests for tools.HTTPMessage

### LINKED ISSUE #242: Add tests for tools.HTTPMessage
Add unit tests for the `tools.HTTPMessage` methods and other functions from [tools/http_message.go](https://github.com/alpine-hodler/gidari/blob/main/tools/http_message.go).

### PR DESCRIPTION
**Pull request by axlts**
Resolves #242

### CODE REVIEW

File: tools/http_message_test.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+		for _, tcase := range []struct {
+			method string
+		}{
+			{method: http.MethodGet},
+			{method: http.MethodPost},
+			{method: http.MethodPut},
+			{method: http.MethodDelete},
+		} {
+			req, err := http.NewRequestWithContext(ctx, tcase.method, "https://someurl.com/path", nil)
```
Comment (prestonvasquez) [inline tools/http_message_test.go:37]: @axlts Can you change all references of "https:// someurl.com" to "https://foo" ? It's a little. That way if (for whatever reason) anyone tries to actually visit the referenced site they'll just get 

> This site can’t be reached

**Review by prestonvasquez — CHANGES_REQUESTED** [review]
Looks great! Just one quick request.

**Comment by axlts** [comment]
@prestonvasquez The build broke, but it looks like it broke because of the connection to db, and not because of my changes.

File: tools/http_message_test.go (hunk 1 above)
Comment (axlts) [reply tools/http_message_test.go]: done

**Review by prestonvasquez — APPROVED** [review]
LGTM! Thank you for this contribution @axlts  :)

**Comment by axlts** [comment]
@prestonvasquez Thanks to you, too! You are the best maintainer I have ever seen!

**Pull request merged**